### PR TITLE
Second batch of static analysis fixes

### PR DESCRIPTION
**Pull request by JelteF**
This fixes some more issues found by static analysis tools. They are separate
from the fixes in #183. Since that PR can almost be merged these are done in a
new PR.

### CODE REVIEW

File: src/bin/pg_autoctl/cli_common.c [hunk 1]
```diff
@@ -494,12 +494,6 @@ cli_create_node_getopts(int argc, char **argv,
 		exit(EXIT_CODE_BAD_ARGS);
 	}
 
-	if (errors > 0)
-	{
-		commandline_help(stderr);
-		exit(EXIT_CODE_BAD_ARGS);
-	}
-
```
Comment (DimCitus) [inline src/bin/pg_autoctl/cli_common.c]: That comes as a surprise, at first re-read of the code I would say we still need this...

File: src/bin/pg_autoctl/pgsql.c [hunk 2]
```diff
@@ -1178,8 +1178,11 @@ pgsql_has_replica(PGSQL *pgsql, char *userName, bool *hasReplica)
 	context.resultType = PGSQL_RESULT_BOOL;
 	context.parsedOk = false;
 
-	pgsql_execute_with_params(pgsql, sql, paramCount, paramTypes, paramValues,
-							  &context, &parseSingleValueResult);
+	if (!pgsql_execute_with_params(pgsql, sql, paramCount, paramTypes, paramValues,
+								   &context, &parseSingleValueResult))
+	{
+		return false;
```
Comment (DimCitus) [inline src/bin/pg_autoctl/pgsql.c]: What about logs? at least a comment saying that `/* errors have already been logged */` after verifying that this is indeed the case?

File: src/bin/pg_autoctl/pgsql.c [hunk 3]
```diff
@@ -1416,8 +1419,11 @@ pgsql_get_postgres_metadata(PGSQL *pgsql, const char *slotName,
 	const char *paramValues[1] = { slotName };
 	int paramCount = 1;
 
-	pgsql_execute_with_params(pgsql, sql, paramCount, paramTypes, paramValues,
-							  &context, &parsePgMetadata);
+	if (!pgsql_execute_with_params(pgsql, sql, paramCount, paramTypes, paramValues,
+								   &context, &parsePgMetadata))
+	{
+		return false;
```
Comment (DimCitus) [inline src/bin/pg_autoctl/pgsql.c]: Same thing with logs.

File: src/bin/pg_autoctl/cli_common.c (hunk 1 above)
Comment (JelteF) [reply src/bin/pg_autoctl/cli_common.c]: This exact check is already done in line 398 and after that errors is not incremented in this function anymore.

**Pull request merged**